pkg/kubelet/active_deadline_test.go

```diff
@@ -42,9 +46,51 @@ func (m *mockPodStatusProvider) GetPodStatus(uid types.UID) (v1.PodStatus, bool)
 	return v1.PodStatus{}, false
 }
 
+func TestNewActiveDeadlineHandler(t *testing.T) {
+	pods := newTestPods(1)
+	podStatusProvider := &mockPodStatusProvider{pods: pods}
+	fakeRecorder := &record.FakeRecorder{}
+	fakeClock := testingclock.NewFakeClock(time.Now())
+
+	testCases := []struct {
+		podStatusProvider status.PodStatusProvider
+		recorder          record.EventRecorder
+		clock             clock.Clock
+	}{
+		{podStatusProvider, fakeRecorder, fakeClock},
+		{podStatusProvider, fakeRecorder, nil},
+		{podStatusProvider, nil, fakeClock},
+		{podStatusProvider, nil, nil},
+		{nil, fakeRecorder, fakeClock},
+		{nil, fakeRecorder, nil},
+		{nil, nil, fakeClock},
+		{nil, nil, nil},
+	}
+
+	for i, testCase := range testCases {
+		actual, err := newActiveDeadlineHandler(testCase.podStatusProvider, testCase.recorder, testCase.clock)
+
+		if i == 0 {
```

Comment:
This appears to only be testing if we can construct an ActiveDeadlineHandler object... Am I missing something?

Reply:
Yep, there is actually only line 54 (which returns error) needs to be covered, but I think that unit test of function NewActiveDeadlineHeader won't be complete without no errors scenario, and I also saw such testing case in other tests

Reply:
I think it's worth adding a comment about that. Something like `0th case is the only one expected to pass, and is kept for coverage and confidence check`

Reply:
@haircommander Done